internal/restic/node.go

```diff
@@ -343,12 +354,23 @@ func (node Node) MarshalJSON() ([]byte, error) {
 
 	type nodeJSON Node
 	nj := nodeJSON(node)
-	name := strconv.Quote(node.Name)
-	nj.Name = name[1 : len(name)-1]
+	nj.Name = quote(node.Name)
+	if nj.LinkTargetQuoted != "" {
+		panic("LinkTargetQuoted must not be set manually")
+	}
+	lt := quote(node.LinkTarget)
+	if lt != node.LinkTarget {
+		// only store if the quoted version is different
+		nj.LinkTargetQuoted = lt
+	}
```

Comment:
Potential optimization:

```suggestion
	if !utf8.ValidString(node.LinkTarget) {
		nj.LinkTargetQuoted = quote(node.LinkTarget)
	}
```

Reply:
This "optimizaion" IMHO only increases the mess, see my comment about non-unicode but escaped filenames below.

Reply:
Whoops, thinko.